app/src/main/java/com/coopra/nebulus/view_models/AlternativeViewModel.java

```diff
@@ -0,0 +1,51 @@
+package com.coopra.nebulus.view_models;
+
+import android.app.Application;
+import android.os.AsyncTask;
+
+import com.coopra.data.Track;
+import com.coopra.nebulus.TrackRepository;
+
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.MutableLiveData;
+
+public class AlternativeViewModel extends AndroidViewModel {
+    private MutableLiveData<List<Track>> mTracks;
+
+    public AlternativeViewModel(@NonNull Application application) {
+        super(application);
+    }
+
+    public MutableLiveData<List<Track>> getTracks() {
+        if (mTracks == null) {
+            mTracks = new MutableLiveData<>();
+        }
+
+        return mTracks;
+    }
+
+    public void retrieveTracks() {
+        new GetTracksTask(getTracks()).execute();
+    }
+
+    private static class GetTracksTask extends AsyncTask<Void, Void, List<Track>> {
```

Comment:
Just fyi....

You can safely make this an inner class that isn't static, since the `ViewModel` doesn't hold a reference to any activities, fragments, views. etc... This will make it simpler, as you don't need to pass in `mTracks` to the constructor and you can just use it directly. However, that doesn't mean this isn't correct. This would work perfectly fine as is, and in someways possibly preferable, in that if this task ever grew more complex, it would be very easily extracted to its own file, since having it static forcing a nice clean break from it parent class.